stage1_fly/run/main.go

```diff
@@ -523,7 +523,7 @@ func stage1(rp *stage1commontypes.RuntimePod) int {
 	}
 
 	diag.Printf("execing %q in %q", args, rfs)
-	if err = syscall.Exec(args[0], args, env); err != nil {
+	if err = syscall.Exec(args[0], args, common.ComposeEnviron(ra.App.Environment)); err != nil {
```

Comment:
See comment above, I think this is dropping the PATH default. I'd suggest to refactor some logic between `WriteEnvFile` and `ComposeEnviron` in environment.go, then here call compose once first and then later use its result for both `WriteEnvFile()` and `Exec()`.